libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -230,6 +231,28 @@ export class Gn4Repository implements RecordsRepositoryInterface {
     )
   }
 
+  openRecordForDuplication(
+    uniqueIdentifier: string
+  ): Observable<[CatalogRecord, string, false] | null> {
+    return this.loadRecordAsXml(uniqueIdentifier).pipe(
+      switchMap(async (xml) => {
+        const converter = findConverterForDocument(xml)
+        const record = await converter.readRecord(xml)
+        return [record, converter] as [CatalogRecord, BaseConverter<string>]
+      }),
+      switchMap(async ([record, converter]) => {
```

Comment:
you can probably leave that aside and have everything in the same `switchMap` :)